### ISSUE TITLE
FunctionDeclaration after ReturnStatement triggers false positive from no-unreachable

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#586]
Consider the following code:

``` js
function foo() {
    return bar();

    function bar() {
        // ...
    }
}
```

Both `no-use-before-define` and `no-unreachable` throw errors here, but I'm of the opinion `no-use-before-define` should be the only error, i.e. `FunctionDeclaration` should be allowed after a `ReturnStatement` in `no-unreachable`.
